### PR TITLE
sample_multinomial and sample_dirichelt_multinomial

### PR DESCRIPTION
**Pull request by kellijohnson-NOAA**
# What is the feature?
* Adds two new functions to sample from the multinomial and the Dirichlet-multinomial distributions.

# How have you implemented the solution?
* Both functions use the multinomial sampling capability in R but the Dirichlet-multinomial does some preprocessing before sampling. The functions are very similar.

# Does the PR impact any other area of the project, maybe another repo?
* Not yet 😁

### CODE REVIEW

File: R/sample_dirichlet_multinomial.R [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  # 1. Sample K probability vectors (p) from a Dirichlet distribution
+  # rdirichlet is often found in packages like 'gtools' or 'MCMCpack'
+  # gtools::rdirichlet(n = 1, alpha = x)
+  # Manually to sample from Dirichlet(alpha)
+  # draw K independent Gamma(alpha_i, 1) variables and normalize by their sum
+  gamma_samples <- stats::rgamma(K, shape = x, rate = 1)
+  dirichlet_probs <- gamma_samples / sum(gamma_samples)
+
+  # 2. Sample counts from a Multinomial distribution, where this is the same
```
Comment (Bai-Li-NOAA) [inline R/sample_dirichlet_multinomial.R]: +1 on calling `sample_multinomial()` here to avoid repetition. If we ever update the multinomial sampling logic, we would only need to do it in one place.

File: R/sample_multinomial.R [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#' @return
+#' A vector the same length as `x` is returned, where the sum of the vector
+#' equals the input to `sample_size`.
+#' @export
+sample_multinomial <- function(x, sample_size) {
+  # Input Validation ----
+  # Check that x is a numeric, positive vector
+  if (!is.numeric(x) || !is.vector(x) || any(x < 0) || length(x) == 0) {
+    cli::cli_abort(c(
+      x = "{.var x} must be a positive numeric vector",
```
Comment (Bai-Li-NOAA) [inline R/sample_multinomial.R]: Do we want to say `{.var x} must be a non-negative numeric vector`? So some of the counts in `x` can be zero.

File: tests/testthat/test-sample_.R [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+  #' @description Test that sample_multinomial(x) returns two different values
+  #' when the seed is not set.
+  expect_false(all(
+    sample_multinomial(c(1, 1), 10) == sample_multinomial(c(1, 1), 10)
+  ))
+})
+
+## Edge handling ----
+test_that("sample_multinomial() returns correct outputs for edge cases", {
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-sample_.R:115]: I wonder if we want to add an additional test case for zeros in `x`. For example: `sample_multinomial(c(1, 0, 5), sample_size = 10)`.

**Review by Bai-Li-NOAA — APPROVED** [review]
The PR looks great, and the documentation is very clear. Thanks, @kellijohnson-NOAA, for working on this!

File: R/sample_multinomial.R (hunk 2 above)
Comment (kellijohnson-NOAA) [reply R/sample_multinomial.R]: Good catch, I will fix this.

Comment (kellijohnson-NOAA) [reply R/sample_multinomial.R]: Fixed in 1d2b107

File: R/sample_dirichlet_multinomial.R (hunk 1 above)
Comment (kellijohnson-NOAA) [reply R/sample_dirichlet_multinomial.R]: done!

File: tests/testthat/test-sample_.R (hunk 3 above)
Comment (kellijohnson-NOAA) [reply tests/testthat/test-sample_.R:115]: I added this test for both dirichlet-multinomial and multinomial, as well as updated the logic for dirichlet-multinomial to allow for one or more but not all zeros, just like the multinomial. I did not see this error in my logic until looking at your suggestions so THANKS 📣👍! The package is much better tested now.
